src/utils/Pikapatible.sol

```diff
@@ -31,31 +35,43 @@ abstract contract Pikapatible is ERC721A, Owned {
         address _settlementContract, 
         address _recipient, 
         uint256 _priceInWei,
-        uint256 _maxSupply
+        uint256 _maxSupply,
+        uint256 _allocatedSupply
     ) Owned(_settlementContract)
     {
         recipient = _recipient;
         price = _priceInWei;
         maxSupply = _maxSupply;
+        allocatedSupply = _allocatedSupply;
     }
 
-    /// @notice May only be called by the Settlement contract
-    /// @dev This mint function can be attached to any ERC721A to enjoy the benefits of the PikaPool auction engine
+    /// @notice May only be called by the Pikapool Settlement contract
+    /// @dev This pikapoolMint function can be attached to any ERC721A via inheritance to enjoy the benefits of the PikaPool auction engine
     /// @dev Will not mint if sent insufficient funds, avoiding reverts to facilitate PikaPool's settlement contract batch minting functionality
     /// @dev ERC721A's _safeMint() function is shirked here in favor of _mint, as all PikaPool mints 
     /// utilize meta-transactions which ensures no smart contracts can bid as they do not possess private keys with which to sign
     /// @param to The bidder address to mint to, provided a sufficient bid was offered
     /// @param amount The number of NFTs to mint to the bidder
-    function mint(address to, uint256 amount) external payable onlyOwner {
-        if (_nextTokenId() + amount > maxSupply) { 
+    function pikapoolMint(address to, uint256 amount) external payable onlyOwner {
+        if (_nextTokenId() + amount > maxSupply || allocatedMints + amount > allocatedSupply) { 
```

Comment:
`allocatedMints` sounds like its the total amount allowed, maybe we could use a name with `count` in it to increase readability?